app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,45 @@
+class Admin::VideosController < ApplicationController
+  before_action :require_sign_in
+  before_action :require_admin
+
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+
+    handel_url
+
+    if @video.save
+      flash[:success] = "You have just created a video!"
+      redirect_to home_path
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def video_params
+    params.require(:video).permit(:title,
+                                  :description,
+                                  :category_id,
+                                  :large_cover,
+                                  :small_cover,
+                                  :url)
+  end
+
+  def require_admin
```

Comment:
I think this should be in `ApplicationController` or in `AdminController` because you will need it in different controllers too. It is not bad for this 1 example.